docusaurus/docs/dev-docs/typescript.md

```diff
@@ -95,6 +93,18 @@ Types can be automatically generated on server restart by adding `autogenerate:
 To use Strapi types in your front-end application, you can [use a workaround](https://github.com/strapi-community/strapi-typed-fronend) until Strapi implements an official solution.
 :::
 
+### Fix build issues with the Generated Types
+
+The generated types can be excluded so that the Entity Service doesn't use them and falls back on looser types that don't check the actual properties available in the content types.
+To do that, edit the `tsconfig.json` of the strapi project and add `types/generated/**` to the `exclude` array.
+
+However, if you still want to use the generated types on your project but don't want Strapi to use them, a workaround could be to copy those generated types and paste them outside of the `generated` directory (so that they
+aren't overwritten when the types are regenerated) and remove the `declare module '@strapi/types'` on the bottom of the file.
+
+:::warning
+Even though `@strapi/types` is exposed to the user, it's for internal use only. The intentionally exposed types for user usage are in `@strapi/strapi`, and we strongly encourage users to use them.
```

Comment:
```suggestion
Types should only be imported from `@strapi/strapi` to avoid breaking changes. The types in `@strapi/types` is for internal use only and is subject to change without notice.
```